### PR TITLE
US137399 Tool filter + interactions

### PR DESCRIPTION
**Pull request by bemailloux**
Tool & role filters are merged together into a single "dimension filter". This PR adds the filter logic as well as interactions - thanks to John's previous role work interactions was like a 6 line change

Quality notes
* Testing
  * Filter
    * Selecting one/multiple/all/clear results in the correct chart update: opacity is applied to TUT and TUC, while filtering is applied to TUR
    * Can search then select items
    * Filter shows all available tools, not just ones that have data for them
    * Both role and tool filter can be applied at the same time
  * Chart interactions
    * Clicking on TUT or TUC applies opacity to TUT/TUC and filters TUR
    * Can select multiple tools at a time
    * Clicking on the last tool selected will clear the filter
  * URL state
    * Updated when filter or charts are updated
  * Export
    * Tool usage export contains only data that satisfies both tool / role filter selections
  * A11y (Chrome x NVDA)
    * Can navigate to each series in TUC and TUT using a combination of Ctrl+up and Ctrl+down
    * To activate a series, "Focus mode" must be on (Insert + spacebar)
    * Filter component has known issues when search is applied - can't go further than a certain point in the list
* Security, devops, docs, perf: N/A
* UX: 
  * [x] Demo

### CODE REVIEW

File: model/dimensionFilter.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		syncWithUrl(this, 'selectedQueryParam', queryParam);
+	}
+
+	get isLoading() {
+		return this._resultSet.isLoading;
+	}
+
+	get items() {
+		if (this._resultSet.isLoading || this._resultSet.isQueryError) return [];
+		return Object.entries(this._getDataFunc(this._resultSet) || {})
```
Comment (johngwilkinson) [inline model/dimensionFilter.js:20]: nit: should be `|| []`

File: model/dimensionFilter.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			.map(([ id, name ]) => ({ id: Number(id), name, isSelected: this._selected.has(Number(id)) }))
+			.sort(compareNameAndId);
+	}
+
+	clear() {
+		this._selected.clear();
+	}
+
+	isSelected(r) {
```
[the remark this answers is not in the document]
Comment (bemailloux) [reply model/dimensionFilter.js:29]: Ahaha I thought I made that change, but it was in an earlier piece of work that I threw away. If there's anything else to update I'll change this along with it.

**Review by johngwilkinson — APPROVED** [review]
Tested on draco, and I just couldn't find anything wrong 😢 
Tested role & tool filters separately & together. Used browser buttons to go back and forth.
Set filters then drilled down and up: filter stayed
Set filters, drilled, changed filters, moved back up using breadcrumb: new filters stayed
Set filters, adjusted date range: filters stayed

On thing which I think is being handled by @bbabu01 but I wanted to mention: adjusting the role filter can change the number of items in the tools legend, and consequently the colours change. Same for the tool filter and the roles legend. Seems like this would be confusing.

**Review by NicholasHallman — APPROVED** [review]
Looks good on demo and draco2

**Pull request merged**